### PR TITLE
Updated stb_image library

### PR DESCRIPTION
**Pull request by sebastianbaginski**
`stb_image` updated to the latest version from official repo.
Needed to add "implementation file" as the library is header-only.
Also updated with our "safety code".
Fixes https://github.com/magic-lang/ooc-kean/issues/1240
@marcusnaslund

### CODE REVIEW

File: source/draw/stb_image/stb_image_impl.c [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#define STB_IMAGE_IMPLEMENTATION
+#define STB_IMAGE_WRITE_IMPLEMENTATION
+
+#include "stb_image.h"
+#include "stb_image_write.h"
+
```
Comment (sebastianbaginski) [inline source/draw/stb_image/stb_image_impl.c]: This file kind of replaces `stb_image.c` and `stb_image_write.c`

**Comment by marcusnaslund** [comment]
Nice. Let's have someone who uses this library peer review. I don't really know who though, @emilwestergren or @davidpiuva ? Also, @fredrikbryntesson , will you do the final review?

**Comment by sebastianbaginski** [comment]
Not much of a review is needed, as our sources are not updated, api is unchanged (and the safety updates are the same as in previous pull requests).
I don't think we need to go through 7K lines of stbi code.

**Comment by marcusnaslund** [comment]
No, I don't mean people should review the C code, just want to know noone complains about updating and/or the implementation.

**Comment by fredrikbryntesson** [comment]
Sure.

**Comment by sebastianbaginski** [comment]
What needs to be done here ?

**Comment by fredrikbryntesson** [comment]
Did the users ok this? @emilwestergren or @davidpiuva or @niklasborwell ?

**Comment by davidpiuva** [comment]
I have my own DebugImage framework so I don't use images so much.

**Comment by sebastianbaginski** [comment]
Yeah I don't think anyone uses this directly, it is only a backend for loading / saving images.

**Comment by marcusnaslund** [comment]
All tests seem to work fine, anyway. Unless someone has complaints today, I'm merging this.

**Comment by marcusnaslund** [comment]
Sorry, there's a conflict in `draw.use`.

**Comment by sebastianbaginski** [comment]
ok, will re-set the flag after rebase

**Comment by sebastianbaginski** [comment]
rebased

**Comment by marcusnaslund** [comment]
Woah, `PRInt` failed.. a lot.

**Comment by sebastianbaginski** [comment]
Hmm bad merge maybe...

**Comment by sebastianbaginski** [comment]
should be ok now

**Comment by marcusnaslund** [comment]
Boom

**Pull request merged**